layer/scripts/update_layer_arn.sh

```diff
@@ -0,0 +1,59 @@
+#!/bin/bash
+
+# This script is run during the reusable_publish_v2_layer_arn CI job,
+# and it is responsible for replacing the layer ARN in our documentation,
+# based on the output files generated by CDK when deploying to each region.
+#
+# see .github/workflows/reusable_deploy_v2_layer_stack.yml
+
+set -eo pipefail
+
+if [[ $# -ne 1 ]]; then
+  cat <<EOM
+Usage: $(basename $0) cdk-output-dir
+
+cdk-output-dir: directory containing the cdk output files generated when deploying the Layer
+EOM
+  exit 1
+fi
+
+CDK_OUTPUT_DIR=$1
+
+# Check if CDK output dir is a directory
+if [ ! -d "$CDK_OUTPUT_DIR" ]; then
+  echo "No $CDK_OUTPUT_DIR directory found, not replacing lambda layer versions"
+  exit 1
+fi
+
+# Process each file inside the directory
+files="$CDK_OUTPUT_DIR/*"
+for file in $files
+do
+  echo "+ $file"
+
+  # Process each line inside the file
+  lines=$(cat "$file")
+  for line in $lines
+  do
+    echo -e "\t- $line"
+    prefix=$(echo "$line" | cut -d ':' -f 1-7)
+
+    # Sed doesn't support \d+ in a portable way, so we cheat with (:digit: :digit: *)
+    sed -i '' -e "s/$prefix:[[:digit:]][[:digit:]]*/$line/g" docs/index.md
+
+    # We use the eu-central-1 layer as the canonical version for all the generic examples
+    # arn:aws:lambda:{region}:017000801446:layer:AWSLambdaPowertoolsPython:39
+    # arn:aws:lambda:${AWS::Region}:017000801446:layer:AWSLambdaPowertoolsPython:39
+    # arn:aws:lambda:${aws:region}:017000801446:layer:AWSLambdaPowertoolsPython:39
+    # arn:aws:lambda:{env.region}:017000801446:layer:AWSLambdaPowertoolsPython:39
+    if [[ "$line" == *"eu-central-1"* ]]; then
+      for region in '{region}' '${AWS::Region}' '${aws:region}' '{env.region}'
```

Comment:
```suggestion
      for pseudo_region in '{region}' '${AWS::Region}' '${aws:region}' '{env.region}'
```

pseudo_region = AWS::Region, etc.